docker-bake.hcl

```diff
@@ -0,0 +1,113 @@
+group "linux" {
+  targets = [
+    "alpine_jdk8",
+    "alpine_jdk11",
+    "debian_jdk8",
+    "debian_jdk11",
+  ]
+}
+
+group "linux-arm64" {
+  targets = [
+    "debian_jdk11",
+  ]
+}
+
+group "linux-s390x" {
+  targets = [
+    "debian_jdk11",
+  ]
+}
+
+group "linux-ppc64le" {
+  targets = [
+    "debian_jdk11",
+  ]
+}
+
+group "windows" {
+  targets = [
+    "windows_2019_jdk11",
+  ]
+}
+
+variable "REMOTING_VERSION" {
+  default = "4.10"
+}
+
+variable "REGISTRY" {
+  default = "docker.io"
+}
+
+variable "JENKINS_REPO" {
+  default = "jenkins/inbound-agent"
+}
+
+variable "BUILD_NUMBER" {
+  default = "1"
+}
+
+variable "ON_TAG" {
+  default = "false"
+}
+
+target "alpine_jdk8" {
+  dockerfile = "8/alpine/Dockerfile"
+  context = "."
+  args = {
+    REMOTING_VERSION = REMOTING_VERSION
+  }
+  tags = [
+    equal(ON_TAG, "true") ? "${REGISTRY}/${JENKINS_REPO}:${REMOTING_VERSION}-${BUILD_NUMBER}-alpine-jdk8": "",
+    "${REGISTRY}/${JENKINS_REPO}:alpine-jdk8",
+    "${REGISTRY}/${JENKINS_REPO}:latest-alpine-jdk8",
+  ]
+  platforms = ["linux/amd64"]
+}
+
+target "alpine_jdk11" {
+  dockerfile = "11/alpine/Dockerfile"
+  context = "."
+  args = {
+    REMOTING_VERSION = REMOTING_VERSION
```

Comment:
this is not the right arg to pass through, the upstream `agent` image includes the remoting version, here we need to calculate the right docker agent version so that we get the right remoting version and pass the `version` arg

🚫 blocker

Reply:
I'm not sure that I'm understanding that part.  The tags on the jenkins/agent image include remoting version plus build number.  The jenkins/inbound-agent image tags include a string that looks like the remoting version plus build number, but it is actually just a copy of the string from the agent image tag.  

I think you're saying that in the inbound-agent we should treat the tag from the jenkins/agent image as the variable, without including a notion of either remoting version or build number.  Have I understood?

Reply:
no the arg passed to the image needs to be overridden at build time otherwise you will get what is the default and it won't actually include the intended remoting version, different images seem to default to different values https://github.com/jenkinsci/docker-inbound-agent/blob/master/11/debian/Dockerfile#L23-L24

this one just would give you the latest alpine image even if you're tagging it with remoting 4.11
https://github.com/jenkinsci/docker-inbound-agent/blob/master/11/alpine/Dockerfile#L24-L25